inference/core/interfaces/stream/inference_pipeline.py

```diff
@@ -146,12 +179,30 @@ def init(
                 disabled in a scenario when Roboflow API key is not given, as Roboflow account is required
                 for this feature to be operational.
-            video_source_properties (Optional[dict[str, float]]): Optional source properties to set up the video source,
-                corresponding to cv2 VideoCapture properties cv2.CAP_PROP_*. If not given, defaults for the video source
-                will be used.
+            video_source_properties (Optional[Union[Dict[str, float], List[Optional[Dict[str, float]]]]]):
+                Optional source properties to set up the video source, corresponding to cv2 VideoCapture properties
+                cv2.CAP_PROP_*. If not given, defaults for the video source will be used.
+                It is optional and if provided can be provided as single dict (applicable for all sources) or
+                as list of configs. Then the list must be of length of `video_reference` and may also contain None
+                values to denote that specific source should remain not configured.
                 Example valid properties are: {"frame_width": 1920, "frame_height": 1080, "fps": 30.0}
             active_learning_target_dataset (Optional[str]): Parameter to be used when Active Learning data registration
                 should happen against different dataset than the one pointed by model_id
+            batch_collection_timeout (Optional[float]): Parameter of multiplex_videos(...) dictating how long process
+                to grab frames from multiple sources can wait for batch to be filled before yielding already collected
+                frames. Please set this value in PRODUCTION to avoid performance drops when specific sources shows
+                unstable latency. Visit `multiplex_videos(...)` for more information about multiplexing process.
+            sink_mode (SinkMode): Parameter that controls how video frames and predictions will be passed to sink
+                handler. With SinkMode.SEQUENTIAL - each frame and prediction triggers separate call for sink,
+                in case of SinkMode.BATCH - list of frames and predictions will be provided to sink, always aligned
+                in the order of video sources - with None values in the place of vide_frames / predictions that
+                were skipped due to `batch_collection_timeout`.
+                `SinkMode.ADAPTIVE` is a middle ground (and default mode) - all old sources will work in that mode
+                against a single video input, as the pipeline will behave as if running in `SinkMode.SEQUENTIAL`.
+                To handle multiple videos - sink needs to accept `predictions: List[Optional[dict]]` and
+                `video_frame: List[Optional[VideoFrame]]`. It is also possible to process multiple videos using
+                old sinks - but then `SinkMode.SEQUENTIAL` is to be used, causing sink to be called on each
+                prediction element.
```

Comment:
I like that we can still use old sinks... Crazy idea: I wonder if there is a way to detect `sink_mode` from the sink based on the type signature?

Reply:
using reflection could possibly get the signature, but I bet people would be making mistakes in that or do not put typing in place - hence I am against this idea. SinkMode adaptive works very reasonable according to explicit input of the pipeline - one vs multiple videos.

Reply:
```python
def b(x, y):
    print(x, y)


from inspect import signature

s = signature(b)

s.parameters["x"].annotation
>>> <class 'inspect._empty'>
```

Reply:
I figured it was a long shot! Thanks for checking.